paasta_tools/chronos_tools.py

```diff
@@ -225,7 +226,17 @@ def get_env(self):
         return [{"name": key, "value": value} for key, value in original_env.iteritems()]
 
     def get_constraints(self):
-        return self.config_dict.get('constraints', [['pool', 'EQUALS', self.get_pool()]])
+        if 'constraints' in self.config_dict:
+            constraints = self.config_dict.get('constraints')
+        else:
+            constraints = self.config_dict.get('extra_constraints', [])
+            constraints.extend(self.get_deploy_constraints())
+            constraints.extend(self.get_pool_constraints())
+        assert isinstance(constraints, list)
+        return [[str(val) for val in constraint] for constraint in constraints]
```

Comment:
This logic needs to be somewhere, and the caller is `format_chronos_job_dict` in this case (and `format_marathon_app_dict` for the corresponding function in MarathonServiceConfig)

I think it's better to have this logic in its own function - `format_chronos_job_dict` is already pretty hairy, and I don't want it getting any longer. Whether that's called `get_constraints` or `combine_constraints` or something is a different argument, but this isn't the only place where we have a `get_something` method that actually looks at other attributes of the config -- e.g. `MarathonServiceConfig.get_instances` is fairly complex.